pkg/consensus/client.go

```diff
@@ -94,27 +103,39 @@ func (c *Client) loadCurrentContext(ctx context.Context, currentSlot types.Slot,
 	return nil
 }
 
+func (c *Client) GetProposerCache(ctx context.Context, slot types.Slot) (ValidatorInfo, bool) {
+	val, ok := c.proposerCache.Get(slot)
+	if !ok {
+		return ValidatorInfo{}, ok
+	}
+	validator, ok := val.(ValidatorInfo)
+	return validator, ok
+}
+
+func (c *Client) GetExecutionCache(ctx context.Context, slot types.Slot) (types.Hash, bool) {
+	val, ok := c.executionCache.Get(slot)
+	if !ok {
+		return types.Hash{}, ok
+	}
+	hash, ok := val.(types.Hash)
+	return hash, ok
```

Comment:
same thing here although no need to return a pointer to a `types.Hash`, can just make the zero value when we need it